src/Bridges/Constraint/norm_to_lp.jl

```diff
@@ -0,0 +1,124 @@
+end
+
+MOI.supports_constraint(::Type{NormInfinityBridge{T}}, ::Type{<:MOI.AbstractVectorFunction}, ::Type{MOI.NormInfinityCone}) where T = true
+MOIB.added_constrained_variable_types(::Type{<:NormInfinityBridge}) = Tuple{DataType}[]
+MOIB.added_constraint_types(::Type{NormInfinityBridge{T}}, ::Type{<:Union{MOI.VectorOfVariables, MOI.VectorAffineFunction{T}}}, ::Type{MOI.NormInfinityCone}) where T = [(MOI.VectorAffineFunction{T}, MOI.Nonnegatives),]
+MOIB.added_constraint_types(::Type{NormInfinityBridge{T}}, ::Type{<:MOI.VectorQuadraticFunction{T}}, ::Type{MOI.NormInfinityCone}) where T = [(MOI.VectorQuadraticFunction{T}, MOI.Nonnegatives),]
+
+# Attributes, Bridge acting as a model
+MOI.get(b::NormInfinityBridge{T}, ::MOI.NumberOfConstraints{<:Union{MOI.VectorOfVariables, MOI.VectorAffineFunction{T}}, MOI.Nonnegatives}) where T = 1
+MOI.get(b::NormInfinityBridge{T}, ::MOI.ListOfConstraintIndices{<:Union{MOI.VectorOfVariables, MOI.VectorAffineFunction{T}}, MOI.Nonnegatives}) where T = [b.nn_index]
+
+# References
+function MOI.delete(model::MOI.ModelLike, c::NormInfinityBridge)
+    MOI.delete(model, c.nn_index)
+end
+
+# Attributes, Bridge acting as a constraint
+function MOI.get(model::MOI.ModelLike, ::MOI.ConstraintPrimal, c::NormInfinityBridge)
+    ge_value = MOI.get(model, MOI.ConstraintPrimal(), c.nn_index)
+    t = sum(ge_value) / length(ge_value)
+    d = div(length(ge_value), 2)
+    x = (ge_value[(d + 1):end] - ge_value[1:d]) / 2
+    return vcat(t, x)
+end
+# Given a_i is dual on t - x_i >= 0 and b_i is dual on t + x_i >= 0,
+# the dual on (t, x) in NormInfinityCone is (u, v) in NormOneCone, where
+# v_i = -a_i + b_i and u = sum(a) + sum(b).
+function MOI.get(model::MOI.ModelLike, ::MOI.ConstraintDual, c::NormInfinityBridge)
+    ge_dual = MOI.get(model, MOI.ConstraintDual(), c.nn_index)
+    t = sum(ge_dual)
+    d = div(length(ge_dual), 2)
+    x = (ge_dual[(d + 1):end] - ge_dual[1:d])
+    return vcat(t, x)
+end
+
+"""
+    NormOneBridge{T}
+
+The `NormOneCone` is representable with LP constraints, since ``t \\ge \\sum_i |x_i|`` if and only if there exists a vector y such that ``t \\ge \\sum_i y_i`` and ``y_i \\ge x_i`` and ``y_i \\ge -x_i`` for all ``i``.
+"""
+struct NormOneBridge{T} <: AbstractBridge
+    y::Vector{MOI.VariableIndex}
+    ge_index::CI{<:Union{MOI.ScalarAffineFunction{T}, MOI.ScalarQuadraticFunction{T}}, MOI.GreaterThan{T}}
+    nn_index::CI{<:Union{MOI.VectorOfVariables, MOI.VectorAffineFunction{T}}, MOI.Nonnegatives}
+end
+function bridge_constraint(::Type{NormOneBridge{T}}, model::MOI.ModelLike, f::MOI.AbstractVectorFunction, s::MOI.NormOneCone) where T
+    if f isa MOI.VectorOfVariables # TODO currently needed for test to pass
+        f = MOI.VectorAffineFunction{T}(f)
+    end
+    f_scalars = MOIU.eachscalar(f)
+    d = MOI.dimension(s)
+    y = MOI.add_variables(model, d - 1)
+    ge_index = MOI.add_constraint(model, MOIU.operate(-, T, f_scalars[1], MOIU.operate(sum, T, y)), MOI.GreaterThan(zero(T)))
```

Comment:
Use MOIU.normalise_and_add_constraint` to move the constant to the set in case it is nonzero